src/SpaceShip.java

```diff
@@ -0,0 +1,50 @@
+public class SpaceShip implements Vehicle{
+
+    private String name;
+
+    private String color;
+
+    private double speed;
+
+    private Brake brake;
+
+    private Engine engine;
+
+    public SpaceShip(String name, String color, Engine engine, Brake brake) {
+        this.name = name;
+        this.speed = 0;
+        this.color = color;
+        this.brake = brake;
+        this.engine = engine;
+    }
+
+    @Override
+    public void accelerate() {
+      speed = speed + engine.getAcceleration() * engine.getAcceleration();
```

Comment:
Qua potrebbe essere meglio salvare l'accelerazione in una variabile temporanea prima, ridurrebbe la lunghezza della linea e ne semplificherebbe la lettura